### PR TITLE
[docs]: add SCALE codec troubleshooting section (#428)

### PR DESCRIPTION
**Pull request by 6r1d**
As discussed with @mversic, we need to add a SCALE codec-related troubleshooting section.

### CODE REVIEW

File: src/guide/troubleshooting/configuration-issues.md [hunk 1]
```diff
@@ -96,3 +96,23 @@ account_ either, unless you submit an instruction.
 These sorts of subtle mistakes can be avoided, for example, by
 deserialising directly from string literals, or by generating a fresh
 key-pair in places where it makes sense.
+
+## SCALE codec errors
+
+You may encounter SCALE codec errors when there's a discrepancy
+between how the data is structured on the client and the peer sides.
```
Comment (WRRicht3r) [inline src/guide/troubleshooting/configuration-issues.md:103]:
```suggestion
between the data structure on the client and the peer sides.
```

File: src/guide/troubleshooting/configuration-issues.md [hunk 2]
```diff
@@ -96,3 +96,23 @@ account_ either, unless you submit an instruction.
...
+
+It's easy to confuse the builds in the `debug` and `release` directories,
+and keep using an old build due to this.
+In case you're using `iroha_client_cli` and rebuild it inside the Iroha
```
Comment (WRRicht3r) [inline src/guide/troubleshooting/configuration-issues.md:107]:
```suggestion
In case you're using the `iroha_client_cli` and rebuild it inside the Iroha
```

File: src/guide/troubleshooting/configuration-issues.md [hunk 3]
```diff
@@ -96,3 +96,23 @@ account_ either, unless you submit an instruction.
...
+project directory root:
```
Comment (WRRicht3r) [inline src/guide/troubleshooting/configuration-issues.md:108]:
```suggestion
project directory root, use these commands:
```

File: src/guide/troubleshooting/configuration-issues.md [hunk 4]
```diff
@@ -96,3 +96,23 @@ account_ either, unless you submit an instruction.
...
+
+* `rm ./target/debug/iroha_client_cli ./target/release/iroha_client_cli`
+* `cargo build -p iroha_client_cli`
+
+Alternatively, please compare the output of
+`/target/debug/iroha_client_cli --version` with a version in the Iroha log.
+You need to view the part at which the Iroha peer starts.
```
Comment (WRRicht3r) [inline src/guide/troubleshooting/configuration-issues.md:115]:
```suggestion
You need to view the part where the Iroha peer starts.
```

File: src/guide/troubleshooting/configuration-issues.md [hunk 5]
```diff
@@ -96,3 +96,23 @@ account_ either, unless you submit an instruction.
...
+This will allow you to see whether the versions match.
+
+If you are using one of our SDKs, please make sure to update it.
```
Comment (WRRicht3r) [inline src/guide/troubleshooting/configuration-issues.md:118]:
```suggestion
If you are using one of our SDKs, please make sure it is up to date.
```

**Review by WRRicht3r — CHANGES_REQUESTED** [review]

File: src/guide/troubleshooting/configuration-issues.md [hunk 6]
```diff
@@ -96,3 +96,23 @@ account_ either, unless you submit an instruction.
...
+You may encounter SCALE codec errors when there's a discrepancy
+between the data structure on the client and the peer sides.
+
+It's easy to confuse the builds in the `debug` and `release` directories,
+and keep using an old build due to this.
+In case you're using the `iroha_client_cli` and rebuild it inside the Iroha
+project directory root, use these commands:
+
+* `rm ./target/debug/iroha_client_cli ./target/release/iroha_client_cli`
+* `cargo build -p iroha_client_cli`
```
Comment (outoftardis) [inline src/guide/troubleshooting/configuration-issues.md:111]: this should be a code block, not a list

File: src/guide/troubleshooting/configuration-issues.md [hunk 7]
```diff
@@ -96,3 +96,23 @@ account_ either, unless you submit an instruction.
...
+
+Alternatively, please compare the output of
+`/target/debug/iroha_client_cli --version` with a version in the Iroha log.
+You need to view the part where the Iroha peer starts.
```
Comment (0x009922) [inline src/guide/troubleshooting/configuration-issues.md:115]: > You need to view the part where the Iroha peer starts.

Which part? Provide an example.

File: src/guide/troubleshooting/configuration-issues.md [hunk 8]
```diff
@@ -96,3 +96,23 @@ account_ either, unless you submit an instruction.
...
+This will allow you to see whether the versions match.
+
+If you are using one of our SDKs, please make sure it is up to date.
```
Comment (0x009922) [inline src/guide/troubleshooting/configuration-issues.md:118]: It should be not necessarily up to date, but the version of SDK should be compatible with the version of Iroha. In order to check it, it is generally recommended to check for the version in the SDKs readme.

File: src/guide/troubleshooting/configuration-issues.md (hunk 6 above)
Comment (0x009922) [inline src/guide/troubleshooting/configuration-issues.md:111]:
```suggestion
* `cargo build -p iroha_client_cli --release`
```

File: src/guide/troubleshooting/configuration-issues.md [hunk 9]
```diff
@@ -96,3 +96,23 @@ account_ either, unless you submit an instruction.
...
+It's easy to confuse the builds in the `debug` and `release` directories,
+and keep using an old build due to this.
+In case you're using the `iroha_client_cli` and rebuild it inside the Iroha
+project directory root, use these commands:
+
+* `rm ./target/debug/iroha_client_cli ./target/release/iroha_client_cli`
+* `cargo build -p iroha_client_cli`
+
+Alternatively, please compare the output of
+`/target/debug/iroha_client_cli --version` with a version in the Iroha log.
```
Comment (0x009922) [inline src/guide/troubleshooting/configuration-issues.md:114]:
```suggestion
`cargo run -p iroha_client_cli --release -- --version` with a version in the Iroha log.
```

File: src/guide/troubleshooting/configuration-issues.md [hunk 10]
```diff
@@ -96,3 +96,23 @@ account_ either, unless you submit an instruction.
 These sorts of subtle mistakes can be avoided, for example, by
 deserialising directly from string literals, or by generating a fresh
 key-pair in places where it makes sense.
+
+## SCALE codec errors
+
+You may encounter SCALE codec errors when there's a discrepancy
+between the data structure on the client and the peer sides.
```
Comment (0x009922) [inline src/guide/troubleshooting/configuration-issues.md:103]: I would rephrase. SCALE codec errors appear when the versions of the data structures differ between the client and the peer sides.

**Review by 0x009922 — CHANGES_REQUESTED** [review]
